### DISCUSSION TITLE
502 Bad Gateway errors

### DISCUSSION THREAD

**Discussion by ChrisHardie** [Pirate-Weather/pirateweather#163] [General]
I hesitate to open a bug report about this because I see other GH issues about it that have seemingly been addressed, so starting a discussion here first. We poll the `https://api.pirateweather.net/forecast` endpoint hourly, and lately we've been seeing pretty regular `502 Bad Gateway` responses from that, for example we've seen eight errors today (as of 12:30 Eastern). The underlying error shared is "could not JSON decode Lambda function response: statusCode validation failed".

Is this a known issue, or is there something we need to do differently? Happy to share any other details that would be useful. Thank you.

**Comment by alexander0042** [comment]
Thanks for posting this here- the one job of an API is to provide accurate data, so I take this stuff seriously! 

One follow-up question that would help me a lot is if is any pattern with these errors? Do retries for the same query give the same error, or is it specific locations or query's?

**Comment by ChrisHardie** [comment]
We're querying the same location each time (query is `/39.827,-84.890?exclude=minutely,hourly,flags`). We are not currently doing an immediate retry because I didn't want to hammer the API if it was a performance or resource issue, so we wait a minimum of 15 minutes. But the same query is successful and returns expected data at other points throughout the day.

Here's a list of timestamps (all Eastern) when that error was returned:

* Today
  * 02:00
  * 03:30
  * 05:00
  * 05:15
  * 05:30
  * 07:00
  * 11:00
  * 12:30
* Yesterday
  * 00:30
  * 02:00
  * 03:30
  * 05:00
  * 06:30
  * 06:45
  * 07:00
  * 08:30
  * 10:00
  * 11:30
  * 15:30
  * 23:15

I hope that helps.

**Comment by pdxvr** [comment]
I'm having this issue too.  I do calls every 20 minutes.  I do 6 requests 

2 of each of these for different locations
extend=hourly&exclude=alerts,daily,minutely,currently
extend=hourly&exclude=alerts,hourly,minutely
exclude=alerts,hourly,minutely

I tend to get the errors on the first 2 requests in my script, (but sometimes it's more than the first 2, but then it's random).  I'm using php and a cron job and tried both curl and file_get_contents with similar results except curl gives me the error message.  Running the php script in the browser seems to always work which seems strange to me.  

Since it's usually the first two, I have considered doing 2 very small api calls first or repeat calls if it fails, but that's not friendly.

**Comment by cloneofghosts** [comment]
I've also on rare occurrences ran into 502 errors but I can refresh immediately and everything works.

**Comment by alexander0042** [comment]
This is on the long list of things that should be fixed in v2! I've traced this back to lambda cold starts, but since v2 doesn't rely on lambda, it shouldn't be an issue there
